django/forms/models.py

```diff
@@ -1125,6 +1125,20 @@ def has_changed(self, initial, data):
         return False
 
 
+class ModelChoiceIteratorValue:
+    def __init__(self, value, instance):
+        self.value = value
+        self.instance = instance
+
+    def __str__(self):
+        return str(self.value)
+
+    def __eq__(self, other):
```

Comment:
Is this tested? It wasn't apparent to me from the changes in the tests.

Reply:
This method was added as, without it, a number of tests fail. Just one example (but not the only) is below. Would you like to see a new test in addition to these?

```
======================================================================
FAIL: test_no_empty_option (forms_tests.tests.tests.ModelFormCallableModelDefault)
If a model's ForeignKey has blank=False and a default, no empty option is created (Refs #10792).
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib64/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/usr/lib64/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/usr/lib64/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/home/jon/devel/django/tests/forms_tests/tests/tests.py", line 79, in test_no_empty_option
    self.assertEqual(choices[0], (option.pk, str(option)))
  File "/usr/lib64/python3.8/unittest/case.py", line 912, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/usr/lib64/python3.8/unittest/case.py", line 1129, in assertTupleEqual
    self.assertSequenceEqual(tuple1, tuple2, msg, seq_type=tuple)
  File "/usr/lib64/python3.8/unittest/case.py", line 1100, in assertSequenceEqual
    self.fail(msg)
  File "/usr/lib64/python3.8/unittest/case.py", line 753, in fail
    raise self.failureException(msg)
AssertionError: Tuples differ: (<django.forms.models.ModelChoiceIteratorV[44 chars]n 1') != (1, 'ChoiceOption 1')

First differing element 0:
<django.forms.models.ModelChoiceIteratorValue object at 0x7f4436c60250>
1

- (<django.forms.models.ModelChoiceIteratorValue object at 0x7f4436c60250>,
-  'ChoiceOption 1')
+ (1, 'ChoiceOption 1')
? +++
```